### ISSUE TITLE
character name

### ISSUE THREAD

**Issue by zartan9** [haroldo-ok/BlocklyVN8bit#19]
hi when using say "text" is there a way to hide "character name" tab above? or have the name change
every time (But itd be easier if i could just hide it. I cant figure out "TEXT" tab linked to picture, so SAY 
i think is best way...

side note: i see you made pazzt too, can bitsy convert to zzt in your other converter?
https://rabbitboots.com/blog/2018/05/04/maze-minders-bitsy-to-zzt/
would "sharp" help it convert to 8bit? https://github.com/lordofduct/BitsySharp

**Comment by haroldo-ok** [comment]
Currently, there is no option to hide the character name, but it shouldn't be very difficult to implement.

Converting to ZZT sounds like a cool idea, and it is looks possible, but the results would be limited, mainly since the character set is not redefinable; anyway, the suggestion has been registered for later (https://github.com/haroldo-ok/bitsy-converter/issues/17)

Using BitsySharp to convert it to 8 bit platforms wouldn't help much, since the language is mostly designed for more robust platforms, and while there exist C# compilers targeting 8 bit platforms, they only implement a subset of the language/API. and BitsySharp probably isn't designed with those limitations in mind.

**Comment by zartan9** [comment]
ah i see! great. Yes please keep me posted about character name hide (or way to change it each time character speaks) as im starting to prototype more and more in blocklyvn8bit :)

**Comment by zartan9** [comment]
also (unrelated to char name, but to text in general) is there a way to have the text and black bg partially show the picture below? like 50 percent transparent? thanks be well.

**Comment by haroldo-ok** [comment]
Hello,

Unfortunately,  the targeted platforms don't have support for translucency.

**Comment by zartan9** [comment]
got it. I can make the jpg with the text included as a work around if need be. :)

**Comment by haroldo-ok** [comment]
That could work, depending on how the image is tailored, but you must keep in mind the color limitations

**Comment by zartan9** [comment]
ah yes thanks for reminder. I like how blocklyvn8bit has a bigger picture background than say "DAAD"
https://github.com/daad-adventure-writer/daad but i notice DAAD has text under...is there a way to
maybe move the text down to bottom in blocklyvn8bit to make the picture look bigger/more visual.
Keep up nice work!

**Comment by haroldo-ok** [comment]
Good idea; it would certainly be nice for the user to be able to configure the visual layout. Added to the "todo" list: https://github.com/haroldo-ok/BlocklyVN8bit/issues/20

**Comment by zartan9** [comment]
for the logic I have it so when user presses 1 goes to a dif bg then if presses 2.. but then the options 1/2 show up again...
how can I progress like if someone presses 1 then the option to press 3 to go to bg3, then 7 to go to bg7 and not go back? thanks
![logic](https://user-images.githubusercontent.com/74879950/147628460-afa3bcbe-ff46-49a9-a62e-6e22432b22a7.png)

**Comment by haroldo-ok** [comment]
Hello; 

In order to prevent the user to go back to a previous choice, it would be necessary to set some variable to identify that the choice was used and put the option inside an "if" block; this would allow to show or hide options dinamically.

There currently isn't an "end" option; one alternative would be to create a block that keeps looping to itself.

As for menus with large amount of options, the application isn't currently capable of handling those.

**Comment by zartan9** [comment]
hi i understand the idea of a loop as a way to kinda end the game in the screen the person is in :)
Ill look into how to see if variables was used. as far as options for large menus, can letter variables be used?
like if condition = a, jump to another scene.. this would give 26 options at least !

**Comment by haroldo-ok** [comment]
The main problem would lie in fitting all the options on the screen; maybe adding some scrolling mechanism to the menu would be a more realistic option.

**Comment by zartan9** [comment]
ah sorry i wasnt clear i mean like every new screen 2 more new options (so only 2 or 3 options per scene)... keep on gaming! :)

**Comment by zartan9** [comment]
building to test prg is showing blue lines for pictures...is there any format i need the pictures to work better?
![blu](https://user-images.githubusercontent.com/74879950/147907922-3d3477c8-f985-46c9-8cc7-be28667665e3.png)
text works fine...

**Comment by haroldo-ok** [comment]
AFAIK, for BlocklyVN8Bit, it must be valid a 320 x 200 PNG image. The 8bitunity library also places further restrictions, but BlocklyVN8Bit should, in theory, automatically convert an generic PNG to the more strict format that 8bitunity requires.

Did some error appear during the build process?

**Comment by zartan9** [comment]
thanks that fixed it, it went to 202 pixels some how! I didnt see any errors... do the images for the portraits have to be smaller/another size? happy new year!

**Comment by haroldo-ok** [comment]
That's weird.. BlocklyVN32x, which BlocklyVN8Bit was based on, used a fixed resolution of 320 x 202; maybe there exists some remaining code that would be still doing that, under some circunstances? That would certainly be a bug.

As for the portraits, BlocklyVN8Bit should be automatically resizing them to fit the screen height, should they become too tall. Of course, there's always the possibility of there being some bug; if you want to resize the portraits manually, you should try to keep them smaller than 200 pixels, but, again, the tool should be resizing them automatically.
